### PR TITLE
Standard currency & balance formatting on pools page + added Swap & Vote menu to navbar

### PR DESCRIPTION
**Pull request by schystz**
## Description of Changes

- HDF-142: Use balancer `formatNumber` function for formatting currencies and token balances throughout the app
- HDF-131: Create a "Swap" tab that when clicked redirects to the balancer swap page
- HDF-132: Create a "Vote" tab that when clicked redirects to our halodao snapshot page

## How to test

Check /pools page and verify that $ prices and HALO & BPT balances are showing up fine.

Swap and Vote tabs/menu added on navbar, and redirecting to correct url 

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-142, https://halodao.atlassian.net/browse/HDF-131, https://halodao.atlassian.net/browse/HDF-132)

### Developer Checklist:

- [x] This PR has a corresponding JIRA ticket
- [x] My branch conforms with our naming convention i.e. `feature/BGM-XXX-description`
- [ ] I have written new tests for your core changes, as applicable
- [x] I have successfully ran tests locally

### Reviewers Checklist:

- [ ] Code is readable and understandable; any unclear parts have explanations
- [ ] Code style confirms to typescript rules
- [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
- [ ] I have successfully ran tests locally

### CODE REVIEW

**Review by tracyarciaga — COMMENTED** [review]
@schystz question for you  ->  screenshot below:
image

File: src/components/Header/index.tsx [hunk 1]
```diff
@@ -327,6 +360,12 @@ export default function Header() {
           <StyledNavLink id={`stake-nav-link`} to={'/vesting'}>
             {t('vesting')}
           </StyledNavLink>
+          <StyledExternalLink id={`swap-nav-link`} href={'https://snapshot.org/#/halodao.eth'}>
+            {t('swap')}
+          </StyledExternalLink>
+          <StyledExternalLink id={`vote-nav-link`} href={'https://balancer.exchange/#/swap'}>
+            {t('vote')}
+          </StyledExternalLink>
```
Comment (tracyarciaga) [inline src/components/Header/index.tsx:366]: baliktad yung link @schystz :)

**Review by tracyarciaga — CHANGES_REQUESTED** [review]

File: src/components/Header/index.tsx (hunk 1 above)
Comment (schystz) [reply src/components/Header/index.tsx:366]: Thanks for the catch @tracyarciaga ! 🤦‍♂️

**Comment by schystz** [comment]
> @schystz question for you -> screenshot below:
> image

For this one, we actually reused `formatNumber` function from balancer code and their rule here is to use 3 decimal digits as default for non-currency formatting:

```
  let format = '0.[000]'
  if (number > 1000) format = '0.[0]a'
  if (number < 1) format = '0.[0000]'
```

File: src/components/Header/index.tsx (hunk 1 above)
Comment (schystz) [reply src/components/Header/index.tsx:366]: This is now resolved @tracyarciaga

**Comment by schystz** [comment]
Thanks for the initial review @tracyarciaga! I believe all your feedback has now been addressed, please let me know should I missed anything or you still need clarification on something.

**Comment by barryhalo** [comment]
@schystz Shouldn't we use the number formatter in pool summary also?

https://github.com/HaloDAO/halodao-interface/blob/4ab99c014518638e1b3834f7607d518501adb1ba/src/components/PoolsSummary/index.tsx#L80

https://github.com/HaloDAO/halodao-interface/blob/4ab99c014518638e1b3834f7607d518501adb1ba/src/components/PoolsSummary/index.tsx#L95

https://github.com/HaloDAO/halodao-interface/blob/4ab99c014518638e1b3834f7607d518501adb1ba/src/components/PoolsSummary/index.tsx#L100

https://github.com/HaloDAO/halodao-interface/blob/4ab99c014518638e1b3834f7607d518501adb1ba/src/components/PoolsSummary/index.tsx#L127

**Comment by schystz** [comment]
> @schystz Shouldn't we use the number formatter in pool summary also?
> 
> https://github.com/HaloDAO/halodao-interface/blob/4ab99c014518638e1b3834f7607d518501adb1ba/src/components/PoolsSummary/index.tsx#L80
> 
> https://github.com/HaloDAO/halodao-interface/blob/4ab99c014518638e1b3834f7607d518501adb1ba/src/components/PoolsSummary/index.tsx#L95
> 
> https://github.com/HaloDAO/halodao-interface/blob/4ab99c014518638e1b3834f7607d518501adb1ba/src/components/PoolsSummary/index.tsx#L100
> 
> https://github.com/HaloDAO/halodao-interface/blob/4ab99c014518638e1b3834f7607d518501adb1ba/src/components/PoolsSummary/index.tsx#L127

Actually we do. Those values from the lines you mentioned are coming from `usePoolsSummary` hook, which already applies the formatting. See https://github.com/HaloDAO/halodao-interface/blob/4ab99c014518638e1b3834f7607d518501adb1ba/src/halo-hooks/usePoolsSummary.ts#L72

**Review by barryhalo — APPROVED** [review]
@schystz Looks good to me!

**Review by tracyarciaga — APPROVED** [review]

**Pull request merged**